### ISSUE TITLE
Support async operations in unpackMessage()

### ISSUE THREAD

**Issue by cameronelliott** [vitalets/websocket-as-promised#34]
I've been experimenting with this library in the browser,
but run into one issue I am not sure how to resolve.

```js
const wsp = new WebSocketAsPromised(wsUrl, {
  packMessage: data => JSON.stringify(data),
 unpackMessage: data => JSON.parse(data)
});
```

For the unpackMessage function, in the browser, I receive a Blob, but I don't think I can get the Blob data before returning from the function. When I tried JSON.parse it just returned undefined.
Maybe data is not a blob in Node, but it seems to be in Chrome.
Let me know if an example would help.

**Comment by vitalets** [comment]
> but I don't think I can get the Blob data before returning from the function

Could you demonstrate this?

**Comment by cameronelliott** [comment]
@vitalets Vitaly, I really love this library, and your work in general, but I am working on a different project right now, so I think it might be a long time before I could get to this.
The main issue as best as I can explain it: (if I am not mistaken)

I think you can read the contents of the Blob inside the function when using Node.js (not sure, I am browser focused)
But in the browser environment, I think, I believe you cannot read the contents of the Blob without awaiting a promise,
which cannot be done inside `unpackMessage`
`Blob.prototype.arrayBuffer()` returns a promise.
`Blob.prototype.stream()` gives a ReadableStream which eventually requires a promise to read the data.
`Blob.prototype.text()` returns a promise.

So, my understanding, as I mid-level JS developer, is that I need to await a promise to get the data from the Blob, in order to process/unpack it, and return it from `unpackMessage`

I hope that helps.
I will close this since I cannot write an example at this time.
Thank you for writing this impressive library.

**Comment by vitalets** [comment]
@cameronelliott I've got the point.
Yes, it seems the library should go more clever way: if `unpackMessage` returns promise - wait for it.
I'll reopen the issue and update the title.

Thanks for your explanation anyway!

**Comment by vitalets** [comment]
Released as v2.0.1. 
Feel free to report any problems!
